### ISSUE TITLE
office files can't open in group folder

### ISSUE THREAD

**Issue by the-matzup** [nextcloud/android#6097]
Office files in group folders (Addon), cannot be opened with the two Office programs (Collabora Online & Only Office). Creating or opening the file leads to an endless waiting time and the error message that it takes too long. 

The problem does not appear in the browser. Even outside the group folders the office programs work inside the Nextcloud Android app. Therefore, the problem is specifically between the Nextcloud app and the group folders. 

Nextcloud Server: 18.0.4
Nextcloud App (3.11.1) on Android (9.1.0.293)

**Comment by the-matzup** [comment]
This error in the app occurs only with users who do not have the rights to share. I found that out for debugging purposes.

**Comment by AndyScherzinger** [comment]
looping in @juliushaertl and @tobiasKaminsky

**Comment by juliushaertl** [comment]
> This error in the app occurs only with users who do not have the rights to share. I found that out for debugging purposes.

Could you give some more details about this? You mean rights to share on the group folder or in general?

Also do you use ACL for the group folder permissions or just the regular ones? Do you see any error in the Nextcloud or Collabora logs?

**Comment by the-matzup** [comment]
I just rechecked it. 

When a group has its permission to share generally taken away, opening the Office documents works. No problem with this setting:
![Bildschirmfoto 2020-05-24 um 14 29 05](https://user-images.githubusercontent.com/47818886/82754294-42555800-9dcc-11ea-9739-a3440677f4ae.png)

The problem occurs when a group or person is denied permission to share in the folder. 
![Bildschirmfoto 2020-05-24 um 14 34 02](https://user-images.githubusercontent.com/47818886/82754338-8b0d1100-9dcc-11ea-94c6-4cb652760a58.png)

I don't have direct access to the log file of Collabora, but there is an error in the Nextcloud log:
`{"reqId":"XdaadsjuXbGYEANqVqEKKAAAAMU","level":3,"time":"2020-05-24T12:32:54+00:00","remoteAddr":"79.xxx.xxx.xx","user":"--","app":"onlyoffice","method":"GET","url":"/index.php/apps/files/directEditing/w8CcZMmHAzcoXMLsa87fNBerYZBrfN6bLxDfsJSKHGR34yoNw7zES3izkQw2XFCk","message":"DirectEditor open: /schueler01.test/files/Office-Test/Doc.docx","userAgent":"Mozilla/5.0 (iPad) Nextcloud-iOS/2.25.9","version":"18.0.4.2","id":"6rvb6a3ac0554"}`

**Comment by AndyScherzinger** [comment]
Pinging @tobiasKaminsky @juliushaertl for feedback

**Comment by juliushaertl** [comment]
I still need to try to reproduce that

**Comment by Jirvil** [comment]
Hello guys!
I ran into the problem described here, but not related to android. It's appear in browser on any platform. I'm using Office Online Integration. I tried to give sharing permissions for the folder and it worked. But I can't give sharing rights to users on all folders. 
I also noticed that this bug only appears with files pre-uploaded by the admin (may be, any other user). If the user uploads the file himself, this problem does not occur even if there are no sharing rights.
Nextcloud log shows this error:
`Error: Undefined array key 0 at /var/www/html/apps/officeonline/lib/Controller/WopiController.php#313`

This part of WopiController.php:
```
312: $userFolder = $this->rootFolder->getUserFolder($wopi->getOwnerUid());
313: $file = $userFolder->getById($fileId)[0];
```
I'm using NC 25.0.5 with Office Online 2.0.1
